pkg/image/admission/imagepolicy/rules/rules.go

```diff
@@ -0,0 +1,143 @@
+package rules
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/image/admission/imagepolicy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+type ImagePolicyAttributes struct {
+	Resource      unversioned.GroupResource
+	Name          imageapi.DockerImageReference
+	OriginalName  string
+	Image         *imageapi.Image
+	ExcludedRules sets.String
+}
+
+type RegistryMatcher interface {
+	Matches(name string) bool
+}
+
+type RegistryNameMatcher imageapi.DefaultRegistryFunc
+
+func (m RegistryNameMatcher) Matches(name string) bool {
+	current, ok := imageapi.DefaultRegistryFunc(m)()
+	if !ok {
+		return false
+	}
+	return current == name
+}
+
+type nameSet []string
```

Comment:
NameSet is small, map isn't better than array of strings.

On Fri, Aug 5, 2016 at 8:48 AM, David Eads <notifications@github.com> wrote:

> In pkg/image/admission/imagepolicy/rules/rules.go
> <https://github.com/openshift/origin/pull/8995#discussion_r73686428>:
>
> > +
> > +type RegistryMatcher interface {
> > +	Matches(name string) bool
> > +}
> > +
> > +type RegistryNameMatcher imageapi.DefaultRegistryFunc
> > +
> > +func (m RegistryNameMatcher) Matches(name string) bool {
> > +	current, ok := imageapi.DefaultRegistryFunc(m)()
> > +	if !ok {
> > +		return false
> > +	}
> > +	return current == name
> > +}
> > +
> > +type nameSet []string
>
> sets.String seems natural here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/f96f5e56ba70c7c0f24f875621efdf1ef45ce3c0#r73686428>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pzIyNGKH-IfXQBf3oPBKyz_SrVgRks5qczEggaJpZM4IlDlb>
> .
>